src/SystemCtl.php

```diff
@@ -70,105 +74,121 @@ public static function unitFromSuffix(string $unitSuffix, string $unitName): Uni
             throw new UnitTypeNotSupportedException('Unit type ' . $unitSuffix . ' not supported');
         }
 
-        return new $unitClass($unitName, new ProcessBuilder([self::$binary]));
+        $commandDispatcher = (new SymfonyCommandDispatcher)
+            ->setTimeout(self::$timeout)
+            ->setBinary(self::$binary);
+
+        return new $unitClass($unitName, $commandDispatcher);
     }
 
     /**
      * List all supported units
      *
      * @param null|string $unitPrefix
-     * @param string[] $unitTypes
+     * @param string[]    $unitTypes
+     *
      * @return array|\string[]
      */
     public function listUnits(?string $unitPrefix = null, array $unitTypes = self::SUPPORTED_UNITS): array
     {
-        $processBuilder = $this->getProcessBuilder()
-            ->add('list-units');
+        $commands = ['list-units'];
```

Comment:
Yeah. I thought this through. And yes, I should have simply stick to the syntax from Symfony here and simply "replace" the hard wired ProcessBuilder. Maybe i will rewind some of the changes an go with your idea with v1.x.

For now i will stop making big changes for 0.x tho